File: client/BillsManager/src/types/bill.ts
export interface Bill {
	id: string
	description: string
	price: number
	dueDate: number
}

export interface BillRequest {
	description: string
	price: number
	dueDate: number
}

export function createBill(billRequest?: BillRequest): BillRequest {
	const bill = {
		description: '',
		price: 0,
		dueDate: 0,
	}

	return Object.assign(bill, billRequest)
}

File: server/Bill.cs
namespace BillsManager;

internal class Bill
{
    public Guid Id { get; private set; }
    public string Description { get; private set; }
    public decimal Price { get; private set; }
    public int DueDate { get; set; }

    public Bill(string description, decimal price, int dueDate, Guid? id = null)
    {
        Id = id ?? Guid.NewGuid();
        Description = description;
        Price = price;
        DueDate = dueDate;
    }
}

